differs/file_diff.go

```diff
@@ -71,3 +70,81 @@ func diffImageFiles(image1, image2 pkgutil.Image) (util.DirDiff, error) {
 	diff, _ = util.DiffDirectory(img1Dir, img2Dir)
 	return diff, nil
 }
+
+type FileLayerAnalyzer struct {
+}
+
+func (a FileLayerAnalyzer) Name() string {
+	return "FileLayerAnalyzer"
+}
+
+// FileDiff diffs two packages and compares their contents
+func (a FileLayerAnalyzer) Diff(image1, image2 pkgutil.Image) (util.Result, error) {
+	var dirDiffs []util.DirDiff
+	// This path to an empty dir will be used for diffing in cases
+	// where one image has more layers than the other
+	emptyPath, err := ioutil.TempDir("", "")
+	if err != nil {
+		return &util.MultipleDirDiffResult{}, err
+	}
+	defer os.RemoveAll(emptyPath)
+
+	// Go through each layer of the first image...
+	for index, layer := range image1.Layers {
+		// ...if there is no corresponding layer in the second image, diff with the empty dir
+		if index >= len(image2.Layers) {
+			diff, err := diffImageFiles(layer.FSPath, emptyPath)
+			if err != nil {
+				return &util.MultipleDirDiffResult{}, err
+			}
+			dirDiffs = append(dirDiffs, diff)
+			continue
+		}
+		// ...else, diff as usual
+		layer2 := image2.Layers[index]
+		diff, err := diffImageFiles(layer.FSPath, layer2.FSPath)
+		if err != nil {
+			return &util.MultipleDirDiffResult{}, err
+		}
+		dirDiffs = append(dirDiffs, diff)
+	}
+
+	// check if there are any additional layers in image2...
+	if len(image2.Layers) > len(image1.Layers) {
+		// ... and diff any additional layers with the empty dir
+		for index := len(image1.Layers); index < len(image2.Layers); index++ {
+			layer2 := image2.Layers[index]
+			diff, err := diffImageFiles(emptyPath, layer2.FSPath)
+			if err != nil {
+				return &util.MultipleDirDiffResult{}, err
+			}
+			dirDiffs = append(dirDiffs, diff)
+		}
+	}
+	return &util.MultipleDirDiffResult{
+		Image1:   image1.Source,
+		Image2:   image2.Source,
+		DiffType: "FileLayer",
+		Diff: util.MultipleDirDiff{
+			DirDiffs: dirDiffs,
+		},
+	}, nil
+}
+
+func (a FileLayerAnalyzer) Analyze(image pkgutil.Image) (util.Result, error) {
+	var directoryEntries [][]pkgutil.DirectoryEntry
+	for _, layer := range image.Layers {
+		layerDir, err := pkgutil.GetDirectory(layer.FSPath, true)
+		if err != nil {
+			return util.FileLayerAnalyzeResult{}, err
+		}
+		directoryEntry := pkgutil.GetDirectoryEntries(layerDir)
```

Comment:
nit: combine with line below so you don't create a new local var

`directoryEntries = append(pkgutil.GetDirectoryEntries(layerDir), directoryEntry)`